src/cloudant/design_document.py

```diff
@@ -86,14 +90,15 @@ def update_view(self, view_name, map_func, reduce_func=None):
         if view is None:
             msg = "View {0} does not exist in this design doc".format(view_name)
             raise CloudantArgumentError(msg)
-        view.map = map_func
-        if reduce_func is not None:
-            view.reduce = reduce_func
+        view = View(self, view_name, map_func, reduce_func, **kwargs)
         self.views.__setitem__(view_name, view)
 
     def delete_view(self, view_name):
         """
         Removes a View from the locally cached DesignDocument View dictionary.
+        Do not use this method if you wish to
+        delete a query index view.  Use
+        :func:`~cloudant.database.CloudantDatabase.delete_index` instead.
```

Comment:
I actually added some more validation to `add_view`, `update_view`, and `delete_view` that raises an exception if a QueryIndexView is found and added more detail to the docstrings as well.  Done in 2707ad460db815447a917e39d8b0d67536a7ea3f.